types/event_buffer.go

```diff
@@ -0,0 +1,41 @@
+package types
+
+const (
+	txEventBufferCapacity = 1000
+)
+
+// Interface assertions
+var _ TxEventPublisher = (*TxEventBuffer)(nil)
+
+// TxEventBuffer is a buffer of events.
+type TxEventBuffer struct {
+	next   TxEventPublisher
+	events []EventDataTx
+}
+
+// NewTxEventBuffer returns a new buffer
+func NewTxEventBuffer(next *EventBus) *TxEventBuffer {
```

Comment:
Maybe we can take the buffer size as an argument since we know how many txs are in the block